transformer_engine/pytorch/attention.py

```diff
@@ -1329,16 +1388,17 @@ def attn_forward_func_with_cp(
         ), f"QKV format of {qkv_format} is not supported with context parallelism!"
     assert(qkv_format != "sbhd" or use_fused_attention
         ), "FlashAttention does not support sbhd format!"
-    assert(not(qkv_format == "thd" and use_fused_attention)
-        ), "FusedAttention does not support thd format!"
-    assert (attn_mask_type in ["causal", "no_mask"]
+    assert (qkv_format != 'thd' or \
+            not use_fused_attention or \
+            attn_mask_type in ["padding", "padding_causal"]
         ), f"Mask type of {attn_mask_type} is not supported with context parallelism!"
-    assert (attn_bias is None or use_fused_attention
-        ), "Attention bias is only supported with FusedAttention!"
+    assert (attn_bias is None or (use_fused_attention and "padding" not in attn_mask_type)
+        ), "Attention bias is only supported with FusedAttention + no_padding!"
```

Comment:
Maybe `"""Attention bias is only supported with FusedAttention and "causal" or "no_mask" mask types!"""`?

Reply:
Done